### DISCUSSION TITLE
Boundary Button that stops at corners

### DISCUSSION THREAD

**Discussion by jbchron** [arkypita/LaserGRBL#1796] [Ideas]
I am relatively new to LaserGRBL and love the product so far. I wanted to know if this button exists. I've used Full Spectrum software and there is a button that runs the boundary of the project, but stops at each corner- click the button again and it goes to the next corner. This is a great option for fine-tuning your work location. Does this exist for LaserGRBL or is there something I can do for this rather than code? A button would be nice.

**Comment by arkypita** [comment]
Each LaserGRBL button (excluded the first 3 buttons) are customizable.
You can put the gcode that you want and use some macro or access to grbl setting.

![image](https://user-images.githubusercontent.com/8782035/159130547-a46b5f54-b4f2-4b34-a67b-a44a61978672.png)

M0 is the command you need. Try something like this:

G0 X[left] Y[bottom]
M3 S[$30*2/100] F4000
M0
G1 Y[top] 
M0
G1 X[right]
M0
G1 Y[bottom] 
M0
G1 X[left]
M0
M5
G0 X0 Y0

**Reply by Tribo-2dot0** [reply]
I used this code to create the button... how would I make it change corners with each click? I would give the first click, it would go to the first corner (upper left) and wait for the next click to change corners.

**Reply by StuartB4** [reply]
You click the little man icon. 

![image](https://user-images.githubusercontent.com/16980026/163281133-db885f42-55c8-4f4e-881b-50f252eabbe9.png)

**Reply by Tribo-2dot0** [reply]
With this code above it does the four corners non-stop (a quick pause on each one), but the Hand and Little Man buttons are not enabled during this process.

**Reply by StuartB4** [reply]
I use the code above and after the first click the man icon comes on, then I click that and the blue cross moves to the next corner and the man comes on again, click again for the next corner etc until it reaches home position. 
This is what is displayed in the left window after it's completed the cycle. 

![image](https://user-images.githubusercontent.com/16980026/163394601-4f05c8a0-6420-4e78-a872-dfc8af9ddbf8.png)

**Reply by StuartB4** [reply]
Are you sure you are not using the code I posted. That just has a delay whithout having to click the man icon.
If you are using that then the P setting may be too low. In the code I posted it is 5 which is 5 seconds delay before moving to the next corner.

**Reply by Tribo-2dot0** [reply]
It does the same as your code but with a shorter time in each corner.

**Reply by Tribo-2dot0** [reply]
I found that this only happens in LaserGrbl Simulator mode (which is on this PC), when it is connected to the laser with grbl 1.1f it works as you said, popping up the little man button and clicking each time you want it to change corner, but it still doesn't work for me because it turns off the laser on pauses... the best option is your code increasing the pause time.

**Reply by StuartB4** [reply]
Yes, I just tried it on the laptop I have my engraver connected to and the laser switches off at the corners when using the code above. 
Must be because the M0 in the code means Program Stop, where as the G04 P code is a Pause or Dwell.

**Comment by arkypita** [comment]
PS: if you double click anywhere in your preview LaserGRBL rapid move your laser head in this position.
I think that this is a better way to fine tuning your piece position.

**Comment by jbchron** [comment]
Thanks so much for the great advice! Did not know a double click would move to that location, but made the button as well and will take it out to my shop to load it up and give it a test. 

Now another question where I may be pushing my luck- Can you zoom in/out in the workspace? Not necessary, but wasn't sure if I was overlooking something.

**Comment by StuartB4** [comment]
This works for me.  
Custom Frame button using a time delay instead of pressing the button again. 
The time is set by the P  (5 seconds.) Change it to what you need. 

G0 X[left] Y[bottom]
M3 S3 F3000
G1 Y[top] 
G04 P5  
G1 X[right]
G04 P5  
G1 Y[bottom] 
G04 P5  
G1 X[left]
G04 P5  
M5
G0 X0 Y0

Also, answer to your other question. No you can't zoom in or out.
